### DISCUSSION TITLE
Creating a variable to track the number of objects detected in a series of images

### DISCUSSION THREAD

**Discussion by matt-humphrey** [ultralytics/yolov5#2473] [Q&A]
I'm developing a model that counts the number of snails on a farm, and have then got a function which will take a static image from google maps and place a marker where each image was taken. I'm trying to link each location/image and the number of snails in that area.

Is it possible to create a variable or a dictionary (or otherwise) which links the number of snails and the image that the count was obtained from, such that I can call that variable after detection?

At the moment, the best method I can think of is using --save-txt when using detect.py, and then parsing through each text file to count the number of snails detected. I'd then create a dictionary with the name of each image as the key, and the number of snails as the value. 

Is there a better way to do this?

Cheers

**Comment by glenn-jocher** [comment]
@matt-humphrey you might want to use YOLOv5 PyTorch Hub models for more customizable python workflows. To count the number of objects in an image:

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

# Images
dir = 'https://github.com/ultralytics/yolov5/raw/master/data/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
results = model(imgs)
results.print()  # or .show(), .save()
```

See PyTorch Hub Tutorial to get started:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://github.com/ultralytics/yolov5/issues/251)
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
